tests/e2e/p/interchain_workflow.go

```diff
@@ -0,0 +1,220 @@
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary/common"
+)
+
+var _ = e2e.DescribeXChain("[Interchain Workflow]", func() {
+	require := require.New(ginkgo.GinkgoT())
+
+	const (
+		transferAmount = 10 * units.Avax
+		weight         = 2_000 * units.Avax // Used for both validation and delegation
+	)
+
+	ginkgo.It("should ensure that funds can be transferred from the P-Chain to the X-Chain and the C-Chain", func() {
+		network := e2e.Env.GetNetwork()
+
+		ginkgo.By("checking that the network has a compatible minimum stake duration", func() {
+			minStakeDuration := cast.ToDuration(network.GetConfig().DefaultFlags[config.MinStakeDurationKey])
+			require.Equal(testnet.DefaultMinStakeDuration, minStakeDuration)
+		})
+
+		ginkgo.By("creating wallet with a funded key to send from and recipient key to deliver to")
+		factory := secp256k1.Factory{}
+		recipientKey, err := factory.NewPrivateKey()
+		require.NoError(err)
+		keychain := e2e.Env.NewKeychain(1)
+		keychain.Add(recipientKey)
+		nodeURI := e2e.Env.GetRandomNodeURI()
+		baseWallet := e2e.Env.NewWallet(keychain, nodeURI)
+		xWallet := baseWallet.X()
+		cWallet := baseWallet.C()
+		pWallet := baseWallet.P()
+
+		ginkgo.By("defining common configuration")
+		recipientEthAddress := evm.GetEthAddress(recipientKey)
+		avaxAssetID := xWallet.AVAXAssetID()
+		// Use the same owner for sending to X-Chain and importing funds to P-Chain
+		recipientOwner := secp256k1fx.OutputOwners{
+			Threshold: 1,
+			Addrs: []ids.ShortID{
+				recipientKey.Address(),
+			},
+		}
+		// Use the same outputs for both X-Chain and C-Chain exports
+		exportOutputs := []*avax.TransferableOutput{
+			{
+				Asset: avax.Asset{
+					ID: avaxAssetID,
+				},
+				Out: &secp256k1fx.TransferOutput{
+					Amt: transferAmount,
+					OutputOwners: secp256k1fx.OutputOwners{
+						Threshold: 1,
+						Addrs: []ids.ShortID{
+							keychain.Keys[0].Address(),
+						},
+					},
+				},
+			},
+		}
+
+		ginkgo.By("adding new node and waiting for it to report healthy")
+		node := e2e.AddEphemeralNode(network, testnet.FlagsMap{})
+		e2e.WaitForHealthy(node)
+
+		ginkgo.By("retrieving new node's id and pop")
+		infoClient := info.NewClient(node.GetProcessContext().URI)
+		nodeID, nodePOP, err := infoClient.GetNodeID(e2e.DefaultContext())
+		require.NoError(err)
+
+		ginkgo.By("adding the new node as a validator", func() {
+			startTime := time.Now().Add(e2e.DefaultValidatorStartTimeDiff)
+			// Validation duration doesn't actually matter to this
+			// test - it is only ensuring that adding a validator
+			// doesn't break interchain transfer.
+			endTime := startTime.Add(30 * time.Second)
+
+			rewardKey, err := factory.NewPrivateKey()
+			require.NoError(err)
+
+			const (
+				delegationPercent = 0.10 // 10%
+				delegationFee     = reward.PercentDenominator * delegationPercent
```

Comment:
I'd rename this similaly to what we have done in [staking rewards test addition](https://github.com/ava-labs/avalanchego/pull/1767)